pareto/utilities/process_data.py

```diff
@@ -0,0 +1,838 @@
+#####################################################################################################
+# PARETO was produced under the DOE Produced Water Application for Beneficial Reuse Environmental
+# Impact and Treatment Optimization (PARETO), and is copyright (c) 2021-2024 by the software owners:
+# The Regents of the University of California, through Lawrence Berkeley National Laboratory, et al.
+# All rights reserved.
+#
+# NOTICE. This Software was developed under funding from the U.S. Department of Energy and the U.S.
+# Government consequently retains certain rights. As such, the U.S. Government has been granted for
+# itself and others acting on its behalf a paid-up, nonexclusive, irrevocable, worldwide license in
+# the Software to reproduce, distribute copies to the public, prepare derivative works, and perform
+# publicly and display publicly, and to permit others to do so.
+#####################################################################################################
+"""
+Module to process data from get_data.py and provide helpful feedback to users if input data is
+insufficient or infeasible.
+
+This module includes error handling for missing data, warnings for missing non-essential data, and
+basic infeasibility testing.
+
+Authors: PARETO Team
+"""
+# Imports
+import warnings
+from pareto.utilities.get_data import (
+    get_valid_input_set_tab_names,
+    get_valid_input_parameter_tab_names,
+)
+from pyomo.environ import Param, units as pyunits
+
+# Returns a list of all valid trucking arcs.
+def get_valid_trucking_arc_list():
+    return [
+        "PCT",
+        "PKT",
+        "PST",
+        "PRT",
+        "POT",
+        "FCT",
+        "CKT",
+        "CST",
+        "CRT",
+        "CCT",
+        "SCT",
+        "SKT",
+        "SOT",
+        "RKT",
+        "RST",
+        "ROT",
+    ]
+
+
+# Returns a list of all valid trucking arcs.
+def get_valid_piping_arc_list():
+    return [
+        "PCA",
+        "PNA",
+        "PPA",
+        "CNA",
+        "CCA",
+        "NNA",
+        "NCA",
+        "NKA",
+        "NSA",
+        "NRA",
+        "NOA",
+        "FCA",
+        "FNA",
+        "RNA",
+        "RCA",
+        "RKA",
+        "RSA",
+        "SNA",
+        "SCA",
+        "SKA",
+        "SRA",
+        "SOA",
+        "ROA",
+    ]
+
+
+# Process the input data (df_sets, df_parameters) from get_data.py.
+# Raise errors and warnings for missing data and infeasibilities.
+def check_required_data(df_sets, df_parameters, config):
+    # Import config options within module to avoid circular module imports
+    from pareto.strategic_water_management.strategic_produced_water_optimization import (
+        WaterQuality,
+        PipelineCost,
+        Hydraulics,
+        PipelineCapacity,
+        InfrastructureTiming,
+    )
```

Comment:
Unfortunately these must be defined within the function, otherwise it results in circular imports (imports from strategic_produced_water..., but strategic also imports from this file). The import here allows imports so the files can be separate which I feel is cleaner.